org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -2703,6 +2753,210 @@ class MarkedString {
+	 */
+	Boolean cancellable
+
+	/**
+	 * Optional, more detailed associated progress message. Contains
+	 * complementary information to the `title`.
+	 *
+	 * Examples: "3/25 files", "project/src/module2", "node_modules/some_dep".
+	 * If unset, the previous progress message (if any) is still valid.
+	 */
+	String message
+
+	/**
+	 * Optional progress percentage to display (value 100 is considered 100%).
+	 * If not provided infinite progress is assumed and clients are allowed
+	 * to ignore the `percentage` value in subsequent in report notifications.
+	 *
+	 * The value should be steadily rising. Clients are free to ignore values
+	 * that are not following this rule.
+	 */
+	Integer percentage;
+}
+
+/**
+ *  The notification payload about progress reporting.
+ * Signaling the end of a progress reporting is done using the following payload:
+ * 
+ * Since 3.15.0
+ */
+@JsonRpcData
+class WorkDoneProgressEnd implements WorkDoneProgressNotification {
+
+	override WorkDoneProgressKind getKind() {
+		WorkDoneProgressKind.end
+	}
+
+	/**
+	 * Optional, a final message indicating to for example indicate the outcome
+	 * of the operation.
+	 */
+	String message
+}
+
+/**
+ *  The base protocol offers also support to report progress in a generic fashion. 
+ *  This mechanism can be used to report any kind of progress including work done progress 
+ *  (usually used to report progress in the user interface using a progress bar) 
+ *  and partial result progress to support streaming of results. 
+ *  A progress notification has the following properties:
+ * 
+ * Since 3.15.0
+ */
+@JsonRpcData
+class ProgressParams {
+
+	Either<String, Number> token
+
+	@JsonAdapter(WorkDoneProgressNotificationAdapter.Factory)
+	WorkDoneProgressNotification value;
+
+	new() {
+	}
+
+	new(@NonNull Either<String, Number> token, WorkDoneProgressNotification value) {
+		this.token = token
+		this.value = value
+	}
+
+}
+
+
+/**
+ *  A parameter literal used to pass a work done progress token.
+ * 
+ * Since 3.15.0
+ */
+interface WorkDoneProgressParams {
+	/**
+	 * An optional token that a server can use to report work done progress.
+	 */
+	def Either<String, Number> getWorkDoneToken()
+	
+	def void setWorkDoneToken(Either<String, Number> token)
+	
+}
+
+/**
+ *  Options to signal work done progress support in server capabilities.
+ * 
+ * Since 3.15.0
+ */
+interface WorkDoneProgressOptions {
+	
+	def Boolean getWorkDoneProgress()
+	
+	def void setWorkDoneProgress(Boolean workDoneProgress)
+	
+}
+
+/**
+ *  A parameter literal used to pass a partial result token.
+ * 
+ * Since 3.15.0
+ */
+interface PartialResultParams {
+	/**
+	 * An optional token that a server can use to report partial results (e.g. streaming) to
+	 * the client.
+	 */
+	def Either<String, Number> getPartialResultToken()
+	
+	def void setPartialResultToken(Either<String, Number> token)
+}
+
+/**
+ *  Abstract class which implements work done progress and partial result request parameter.
```

Comment:
Looks like some of the comment formatting is a little off. Not super important. Autoformat does create too much noise I suppose?